src/services/codefixes/addNameToNamelessParameter.ts

```diff
@@ -26,24 +33,43 @@ registerCodeFix({
     getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => makeChange(changes, diag.file, diag.start)),
 });
 
-function makeChange(changeTracker: textChanges.ChangeTracker, sourceFile: SourceFile, pos: number) {
-    const token = getTokenAtPosition(sourceFile, pos);
+function makeChange(changeTracker: textChanges.ChangeTracker, sourceFile: SourceFile, start: number) {
+    const token = getTokenAtPosition(sourceFile, start);
     const param = token.parent;
     if (!isParameter(param)) {
         return Debug.fail("Tried to add a parameter name to a non-parameter: " + Debug.formatSyntaxKind(token.kind));
     }
 
-    const i = param.parent.parameters.indexOf(param);
+    const parameters = param.parent.parameters;
+    const i = parameters.indexOf(param);
     Debug.assert(!param.type, "Tried to add a parameter name to a parameter that already had one.");
     Debug.assert(i > -1, "Parameter not found in parent parameter list.");
 
-    const typeNode = factory.createTypeReferenceNode(param.name as Identifier, /*typeArguments*/ undefined);
+    let end = param.name.getEnd();
+    let typeNode: TypeNode = factory.createTypeReferenceNode(param.name as Identifier, /*typeArguments*/ undefined);
+    let nextParam = tryGetNextParam(sourceFile, param);
+    while (nextParam) {
+        typeNode = factory.createArrayTypeNode(typeNode);
+        end = nextParam.getEnd();
+        nextParam = tryGetNextParam(sourceFile, nextParam);
+    }
+
     const replacement = factory.createParameterDeclaration(
         param.modifiers,
         param.dotDotDotToken,
         "arg" + i,
         param.questionToken,
-        param.dotDotDotToken ? factory.createArrayTypeNode(typeNode) : typeNode,
-        param.initializer);
-    changeTracker.replaceNode(sourceFile, param, replacement);
+        param.dotDotDotToken ? isArrayTypeNode(typeNode) ? typeNode : factory.createArrayTypeNode(typeNode) : typeNode,
+        param.initializer,
+    );
+    changeTracker.replaceRange(sourceFile, createRange(param.getStart(sourceFile), end), replacement);
+}
+
+function tryGetNextParam(sourceFile: SourceFile, param: ParameterDeclaration) {

```

Comment:
```suggestion
/**
 * If a user wrote
 *
 *     (string[][]) => string
 *
 * the parser will see this as an erroneously written
 *
 *     (string, [], []) => string
 *
 * Here we effectively reparse and return these binding pattern parameters
 * as long as we have subsequent '[]' pairs.
 */
 function tryGetNextArrayParam(sourceFile: SourceFile, param: ParameterDeclaration) {
```

Reply:
Do I understand correctly that this QF should handle only the following cases?

1. `(...string)` => `(...arg: string[])`
2. `(...string[])` =>  `(...arg: string[])`

Should the case with `...string[][]` be converted to `(...arg0: string[], [])` or `(...arg0: string[], arg1: string[])`?

Reply:
I think the code is correct as-is; I wouldn't limit it to 1 trailing `[]`.
(I didn't read @DanielRosenwasser 's comment as saying that it should be limited to 1.)

Reply:
Yeah, I'm just trying to clarify the logic with the comment